src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -131,6 +131,7 @@ internal void SetWellKnownType(WellKnownType wellKnownType)
         private TypeFlags InitializeTypeFlags(TypeFlags mask)
         {
             TypeFlags flags = ComputeTypeFlags(mask);
+            flags = Context.ComputeTypeFlags(this, flags, mask);
```

Comment:
Would it make sense to condition this on `(flags & mask) == 0`? This would save two virtual calls and more.